### PR TITLE
[fix](cloud-mow) get delete bitmap update lock should create a new Transaction when encounter TXN_TOO_OLD

### PR DESCRIPTION
**Pull request by hust-hhb**
### What problem does this PR solve?

Related PR: #37670

Problem Summary:
When the number of tablet is big, get delete bitmap update lock may cost over 5 seconds, which will lead to transaction timeout, we should create a new transaction to get the remaining tablet stats.

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by hust-hhb** [comment]
run buildall

**Review by zhannngchen — DISMISSED** [review]
LGTM

File: cloud/src/meta-service/meta_service_tablet_stats.cpp [hunk 1]
```diff
@@ -36,7 +36,8 @@
 
 namespace doris::cloud {
 
-void internal_get_tablet_stats(MetaServiceCode& code, std::string& msg, Transaction* txn,
+void internal_get_tablet_stats(MetaServiceCode& code, std::string& msg,
+                               std::unique_ptr<Transaction>& txn, std::shared_ptr<TxnKv>& txn_kv,
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_tablet_stats.cpp:40]: This is a behavior change, semantic changed, txn atomic is not garanteed if we create a new txn.
Fix where this `internal_get_tablet_stats()` is called if we met  TXN_TOO_OLD

**Review by gavinchou — CHANGES_REQUESTED** [review]

File: cloud/src/meta-service/meta_service_tablet_stats.cpp (hunk 1 above)
Comment (hust-hhb) [reply cloud/src/meta-service/meta_service_tablet_stats.cpp]: @gavinchou already change the code, take a look please

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

File: cloud/src/meta-service/meta_service.cpp [hunk 2]
```diff
@@ -2124,10 +2124,30 @@ void MetaServiceImpl::get_delete_bitmap_update_lock(google::protobuf::RpcControl
...
             TabletStatsPB tablet_stat;
+            int64_t retry = 0;
             internal_get_tablet_stats(code, msg, txn.get(), instance_id, idx, tablet_stat, false);
+            count++;
+            TEST_SYNC_POINT_CALLBACK("get_tablet_stats_code", &count, &code);
+            while (retry < 3 && code == MetaServiceCode::KV_TXN_TOO_OLD) {
+                code = MetaServiceCode::OK;
+                txn = nullptr;
+                retry++;
+                err = txn_kv_->create_txn(&txn);
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service.cpp:2139]: overwrite the `txn` cannot modify the "delete_bitmap_upldate_lock"

**Pull request closed without merging**